vm/src/format.rs

```diff
@@ -535,8 +535,7 @@ impl FormatSpec {
     ) -> Result<String, &'static str> {
         let align = self.align.unwrap_or(default_align);
 
-        // Use the byte length as the string length since we're in ascii
-        let num_chars = magnitude_string.len();
+        let num_chars = magnitude_string.chars().count();
```

Comment:
ah, this is not the best. probably we'd better to pass StringKind here to not to count characters for ascii strings.

Reply:
Code fix completed.